### DISCUSSION TITLE
How can I query the list via a id string?

### DISCUSSION THREAD

**Discussion by yaoqiangpersonal** [querydsl/querydsl#3121] [Q&A] [answered]
`@Entity
public class Demo {

    @Id
    private Integer id;

    private String name;
}
`
I have an entity like this. I want to get items that id is 1 or 2 by sending params '?idCollection=1,2'. What could I do? I use springboot data jpa.

**Comment by zhanglolo** [comment, marked as answer]
maybe you can try this:
1. use `@RequestParam` get `idCollection` then conver `List`, like `List<String> itemIds`.
2. write `jpaQueryFactory.selectFrom(qDemo).where(qDemo.id.in(itemIds))` which means `select * from demo where id in ('1', '2')`

**Reply by yaoqiangpersonal** [reply]
thanks for your reply. I have used the @QuerydslPredicate, so the query params have to equal to the entity. In the past, I wanted to use comma cutting to get more detailed parameters. finally, I found it's not a good idea, i changed the type of method that send params like `"A=xx&A=bb"`, then A can be converted `MutiValueBingding`, In the `QuerydslPredicateExecutor`, config `QuerydslBindings.bind(root.A).all(path,value)->Optional.of(path.in(value)))`, it's the same result like yours.
